### ISSUE TITLE
Deepcopy not working 

### ISSUE THREAD

**Issue by guardeivid** [geographika/mappyfile#73]
Hi, I tried to use **`deepcopy`** and the following error occurs:

```python
from copy import deepcopy
from mappyfile.ordereddict import DefaultOrderedDict

d = DefaultOrderedDict()
d['__type__'] = 'layer'

e = deepcopy(d)
print(e)
```

```error
Traceback (most recent call last):
  File "C:\OSGeo4W\apps\Python36\lib\code.py", line 91, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "<string>", line 18, in <module>
  File "C:\OSGeo4W\apps\Python36\lib\copy.py", line 161, in deepcopy
    y = copier(memo)
  File "C:\OSGeo4W\apps\Python36\lib\site-packages\mappyfile\ordereddict.py", line 61, in __deepcopy__
    copy.deepcopy(self.items()))
  File "C:\OSGeo4W\apps\Python36\lib\copy.py", line 169, in deepcopy
    rv = reductor(4)
TypeError: can't pickle odict_items objects
```

This is happening in Python >= 3.5

Following the logic maybe it should be like this?
```python
# ordereddict.py

    def __deepcopy__(self, memo):
        return type(self)(self.default_factory, copy.deepcopy(iter(self.items())))
```

**Comment by geographika** [comment]
@guardeivid - thanks for the report and suggested fix. Should be fixed with the commit - df590b536d0742dce4d1308f0da7b2d15a147b00

I had to use list() rather than iter() or Python2 breaks. Unit test also added. This will be part of the next patch release.

**Comment by guardeivid** [comment]
Thanks
